### PR TITLE
Script to verify the upcoming UMIP-3 DVM upgrade

### PR DESCRIPTION
**Pull request by chrismaree**
This script assumes you've run both the `Propose.js` followed by the `VoteSimulator.js` script to get the DVM to an upgraded state. This script aims to ensure that all permissioning and state changes are done correctly post upgrade.

### CODE REVIEW

**Comment by chrismaree** [comment]
The next thing to do with this script is run some basic DVM logic like submitting price requests, voting on prices and resolving prices. Perhaps running another upgrade following this upgrade.

File: core/scripts/umip-3/Verify.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  FinancialContractsAdmin: "0x3b99859be43d543960803c09a0247106e82e74ee",
+  IdentifierWhitelist: "0x9e39424eab9161cc3399d886b1428cba71586cb8",
+  Governor: "0x878cfedb234c226ddefd33657937af74c17628bf",
+  Finder: "0x40f941E48A552bF496B154Af6bf55725f18D77c3" // Finder was no upgraded in UMIP3
+};
+
+async function runExport() {
+  const finder = await Finder.deployed();
+  const oldVoting = await Voting.deployed();
+  console.log(oldVoting.address);
```
Comment (mrice32) [inline core/scripts/umip-3/Verify.js:38]: Did you mean to remove this? Doesn't seem to be used.

File: core/scripts/umip-3/Verify.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  assert.equal(onChainByteCode, compiledByteCode);
+};
+
+// Ensure that the finder has the correct contract address for a given interface name
+finderMatchesDeployment = async (contract, interfaceName) => {
+  const finder = await Finder.deployed();
+  const interfaceNameBytes32 = web3.utils.utf8ToHex(interfaceName);
+  const finderImplementationAddress = await finder.getImplementationAddress(interfaceNameBytes32);
+  const upgradeDeployedAddress = upgradeAddresses[contract.contractName];
+  assert.equal(finderImplementationAddress.toLowerCase(), upgradeDeployedAddress.toLowerCase());
```
Comment (mrice32) [inline core/scripts/umip-3/Verify.js:126]: nit: prefer `web3.utils.toChecksumAddress(address)`

File: core/scripts/umip-3/Verify.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+};
+
+// Ensure that a given contract is owned by the NewGovernor
+contractOwnedByNewGovernor = async contract => {
+  const contractInstance = await contract.at(upgradeAddresses[contract.contractName]);
+  const currentOwner = await contractInstance.owner();
+  assert.equal(currentOwner.toLowerCase(), upgradeAddresses.Governor.toLowerCase());
```
Comment (mrice32) [inline core/scripts/umip-3/Verify.js:133]: Same as above.

File: core/scripts/umip-3/Verify.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+};
+
+// Ensure that a given contract's multirole for `owner` is set to the new GovGovernor
+newGovernorHasOwnerRole = async contract => {
+  const contractInstance = await contract.at(upgradeAddresses[contract.contractName]);
+  const exclusiveRoleHolder = await contractInstance.getMember(ownerRole);
+  assert.equal(exclusiveRoleHolder.toLowerCase(), upgradeAddresses.Governor.toLowerCase());
```
Comment (mrice32) [inline core/scripts/umip-3/Verify.js:140]: Same as above.

File: core/scripts/umip-3/Verify.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+};
+
+// Ensure that a given contract's multirole for `owner` is set to the foundation multisig wallet
+contractOwnerRoleByFoundation = async contract => {
+  const contractInstance = await contract.at(upgradeAddresses[contract.contractName]);
+  const exclusiveRoleHolder = await contractInstance.getMember(ownerRole);
+  assert.equal(exclusiveRoleHolder.toLowerCase(), foundationWallet.toLowerCase());
```
Comment (mrice32) [inline core/scripts/umip-3/Verify.js:147]: Same as above.

**Pull request merged**